crates/node/src/connection.rs

```diff
@@ -10,10 +10,10 @@ pub struct NodeClient {
     /// Note: this is an [`Option`] *only* to satisfy the borrow checker. It’s
     /// *always* [`Some`]. See [`<super::pool_manager::NodePoolManager as
     /// deadpool::managed::Manager>>::recycle`] for an explanation.
-    pub(in crate::node) client: Option<NodeClientFacade>,
-    pub(in crate::node) connection_id: u64,
-    pub(in crate::node) unrecoverable_error_happened: bool,
-    pub(in crate::node) network_magic: u64,
```

Comment:
Why this change? 👀 We lose the safe encapsulation, if we expose everything publicly.